web/modules/custom/joinup_community_content/src/Controller/PinContentController.php

```diff
@@ -0,0 +1,182 @@
+   *   The node entity being unpinned.
+   *
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   The redirect response.
+   */
+  public function unpin(NodeInterface $node) {
+    $node->setSticky(FALSE);
+    $node->save();
+
+    drupal_set_message($this->t('@bundle %title has been unpinned in the collection %collection.', [
+      '@bundle' => $node->get('type')->entity->label(),
+      '%title' => $node->label(),
+      '%collection' => $this->relationManager->getParent($node)->label(),
+    ]));
+
+    return $this->getRedirect($node);
+  }
+
+  /**
+   * Access check for the pin route.
+   *
+   * A node can be pinned only if it's not pinned, if its bundle is a community
+   * content one and if the user is a facilitator in the parent collection.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node entity being pinned.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user account to check access for.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result.
+   */
+  public function pinAccess(NodeInterface $node, AccountInterface $account) {
+    return AccessResult::allowedIf(
+      !$node->isSticky() &&
+      in_array($node->bundle(), CommunityContentHelper::getBundles()) &&
+      $this->isFacilitatorInParentCollection($node, $account)
+    );
+  }
+
+  /**
+   * Access check for the unpin route.
+   *
+   * A node can be unpinned only if it's pinned, if its bundle is a community
+   * content one and if the user is a facilitator in the parent collection.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node entity being unpinned.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The user account to check access for.
+   *
+   * @return \Drupal\Core\Access\AccessResult
+   *   The access result.
+   */
+  public function unpinAccess(NodeInterface $node, AccountInterface $account) {
+    return AccessResult::allowedIf(
+      $node->isSticky() &&
+      in_array($node->bundle(), CommunityContentHelper::getBundles()) &&
+      $this->isFacilitatorInParentCollection($node, $account)
+    );
+  }
+
+  /**
+   * Checks if a user has facilitator role in the parent collection of a node.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node entity.
+   * @param \Drupal\Core\Session\AccountInterface $account
+   *   The account to check the OG role for.
+   *
+   * @return bool
+   *   True if the user has the facilitator role in the parent of the node.
+   */
+  protected function isFacilitatorInParentCollection(NodeInterface $node, AccountInterface $account) {
+    $collection = $this->relationManager->getParent($node);
+
+    if (!$collection || $collection->bundle() !== 'collection') {
+      return FALSE;
+    }
+
+    $membership = $this->membershipManager->getMembership($collection, $account);
+    return !empty($membership) && $membership->hasRole('rdf_entity-collection-facilitator');
```

Comment:
Oh, no! Again and again we are repeating this wrong way to check the access. We should never ever, but never, check against the role. Starting from version 4 (or even 3), Drupal has introduced this layer or permissions that allows a higher flexibility. For example, with the actual code, the super-admin (uid1) is not able to pin/unpin content because he's not a group facilitator. But if a permission is checked, rather than a role ID, super-admin has, by default, all the permissions even are not granted explicitly.

So here we need to:
- Create a new collection OG permission: `pin community content` (we can use this permission also for unpinning).
- Grant this permission to collection owner and facilitator OG roles.
- Check here against the new permission. I think `\Drupal::service('og.access')->userAccess($collection, 'pin community content')` is doing the job (but injected). Rename this function to reflect the changes.